tests/appsec/test_automated_login_events.py

```diff
@@ -0,0 +1,175 @@
+    BASIC_AUTH_USER_UUID_HEADER = "Basic dGVzdHV1aWQ6MTIzNA=="  # base64(testuuid:1234)
+    BASIC_AUTH_INVALID_USER_HEADER = "Basic aW52YWxpZFVzZXI6MTIzNA=="  # base64(invalidUser:1234)
+    BASIC_AUTH_INVALID_PASSWORD_HEADER = "Basic dGVzdDoxMjM0NQ=="  # base64(test:12345)
+
+    def setup_login_pii_success(self):
+        self.library_name = context.library
+        self.r_pii_success = []
+
+        if self.library_name == "nodejs":
+            self.r_pii_success = [
+                weblog.post("/login?auth=local", data={"username": self.USER, "password": self.PASSWORD}),
+                weblog.get("/login?auth=basic", headers={"Authorization": self.BASIC_AUTH_USER_HEADER}),
+            ]
+
+    def test_login_pii_success(self):
+        for r in self.r_pii_success:
+            assert r.status_code == 200
+            for _, _, span in interfaces.library.get_spans(request=r):
+                meta = span.get("meta", {})
+                assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "safe"
+                assert meta["appsec.events.users.login.success.track"] == "true"
+                assert meta["usr.id"] == " "
+                assert meta["manual.keep"] == "true"
+
+    def setup_login_success(self):
+        self.library_name = context.library
+        self.r_success = []
+
+        if self.library_name == "nodejs":
+            self.r_success = [
+                weblog.post("/login?auth=local", data={"username": self.UUID_USER, "password": self.PASSWORD},),
+                weblog.get("/login?auth=basic", headers={"Authorization": self.BASIC_AUTH_USER_UUID_HEADER}),
+            ]
+
+    def test_login_success(self):
+        for r in self.r_success:
+            assert r.status_code == 200
+            for _, _, span in interfaces.library.get_spans(request=r):
+                meta = span.get("meta", {})
+                assert meta["_dd.appsec.events.users.login.success.auto.mode"] == "safe"
+                assert meta["appsec.events.users.login.success.track"] == "true"
+                assert meta["usr.id"] == "591dc126-8431-4d0f-9509-b23318d3dce4"
+                assert meta["manual.keep"] == "true"
+
+    def setup_login_wrong_user_failure(self):
+        self.library_name = context.library
+        self.r_wrong_user_failure = []
+
+        if self.library_name == "nodejs":
+            self.r_wrong_user_failure = [
+                weblog.post("/login?auth=local", data={"username": self.INVALID_USER, "password": self.PASSWORD}),
+                weblog.get("/login?auth=basic", headers={"Authorization": self.BASIC_AUTH_INVALID_USER_HEADER}),
+            ]
+
+    def test_login_wrong_user_failure(self):
+        for r in self.r_wrong_user_failure:
+            assert r.status_code == 401
+            for _, _, span in interfaces.library.get_spans(request=r):
+                meta = span.get("meta", {})
+                if self.library_name != "nodejs":
```

Comment:
Can you comment here why the behavior of nodejs is different from other tracers ?